samvera-labs/clover-iiif
# Add ESLint / Prettier config

## What does this do?
- Creates and wires up the `lint` command (hooking into NextJS's `lint`)
  - Uses lint rules from `@typescript-eslint/recommended`
- Adds `prettier` as a dev dependency to the project and setup command line utils (`prettier` and `prettier:fix`)
- Syncs `prettier` package to `eslint` package
- Make linting / typechecking part of Clover's Continuous Integration via `lint-staged` package.  It will run on every `git commit`.
- Change TypeScript command line util naming to `typecheck` (previously `ts-lint`)

## How to test?
Create a test file in the project, ie, `test.ts` and make it intentionally violate rules.  For example

```ts
let foo;
export const bar: number = "baz";
```

Save the file, git stage the file and try to create a new commit.  You will see the linting run and your commit should be blocked.

![image](https://github.com/samvera-labs/clover-iiif/assets/3020266/6cfabc00-9702-4413-9045-9bfff01c5926)


```shell
# Run Prettier checks
pnpm prettier

# Run Prettier fix on poorly formatted code
pnpm pretter:fix

# Run ESLint
pnpm lint
```
